Fix for issue #26

sumeetchhetri opened the pull request:
Register a receiver level parser exception handler to receive notifications
about any protocol level parsing errors

sumeetchhetri commented:
@jamesagnew, please have a look at this.

jamesagnew commented:
Hi @sumeetchhetri - This looks great, thanks for the contribution.

Would it be possible to add a test?

sumeetchhetri commented:
@jamesagnew, had a horrific experience trying to setup the local dev environment for the project, added the test case but unable to validate locally due to lots of maven related issues. Added the test case as requested.
